crates/router/src/connector/stax/transformers.rs

```diff
@@ -85,11 +185,32 @@ impl From<StaxPaymentStatus> for enums::AttemptStatus {
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum StaxPaymentResponseTypes {
+    Charge,
+    PreAuth,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct StaxChildCapture {
+    id: String,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq)]
 pub struct StaxPaymentsResponse {
-    status: StaxPaymentStatus,
+    success: bool,
     id: String,
+    is_captured: i8,
+    is_voided: bool,
+    child_captures: Vec<StaxChildCapture>,
+    #[serde(rename = "type")]
+    payment_response_type: StaxPaymentResponseTypes,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct StaxCaptureId {
```

Comment:
Also, please add a comment which id it is, because now it mentioned in struct name as capture id but when u change it to "StaxMetaData", it won't be clear which id it is

Reply:
Done